widgets/filters/src/FilterModal.tsx

```diff
@@ -0,0 +1,69 @@
+import { FunctionComponent } from 'react';
+import { Filter } from './types';
+
+export interface FilterModalProps {
+    initialFilter?: Filter;
+    filterId?: string;
+    i18nHeaderKey: string;
+    onSubmit: (filterId: string, filterRules: []) => Promise<void>;
+    onCancel: () => void;
+}
+
+const { Form } = Stage.Basic;
+
+// TODO: RD-1837
+const UnsafeForm = (Form as unknown) as FunctionComponent<{ [x: string]: any }>;
+const UnsafeField = (Form.Field as unknown) as FunctionComponent<{ [x: string]: any }>;
+
+const FilterModal: FunctionComponent<FilterModalProps> = ({
+    i18nHeaderKey,
+    initialFilter,
+    filterId: filterIdProp,
+    onCancel,
+    onSubmit
+}) => {
+    const { useInput, useErrors } = Stage.Hooks;
+    const [filterId, setFilterId] = useInput(filterIdProp);
+    const { errors, setErrors, clearErrors, setMessageAsError } = useErrors();
+    const { i18n } = Stage;
+
+    function handleSubmit() {
+        clearErrors();
+
+        if (filterIdProp !== undefined && !filterId) {
+            setErrors({ filterId: i18n.t('widgets.filters.modal.idMissing') });
+            return;
+        }
+
+        onSubmit(
+            (filterId || initialFilter?.id) as string,
+            initialFilter ? [...initialFilter.attrs_filter_rules, ...initialFilter.labels_filter_rules] : []
```

Comment:
Optional:

How about extracting the logic that combines `attrs_filter_rules` and `labels_filter_rules` into a separate function? e.g. `getConcatenatedFilterRules`

```suggestion
            initialFilter ? getConcatenatedFilterRules(initialFilter) : []
```

Reply:
This is temporary - once rules form will be in place the list of rules to be submitted will be provided by the form.

Reply:
Makes sense :+1: